pkg/windows/utils.go

```diff
@@ -101,15 +102,35 @@ func deleteAllNetworks() error {
 		return err
 	}
 
+	var ips []string
+
 	for _, network := range networks {
 		if network.Name != "nat" {
+			logrus.Debugf("Deleting network: %s before starting calico", network.Name)
+			ips = append(ips, network.ManagementIP)
 			_, err = network.Delete()
 			if err != nil {
 				return err
 			}
 		}
 	}
 
+	// HNS overlay networks restart the physical interface when they are deleted. Wait until it comes back before returning
+	waitErr := wait.Poll(2000*time.Millisecond, 20*time.Second, func() (bool, error) {
```

Comment:
it seems `wait.Poll` is deprecated: https://pkg.go.dev/k8s.io/apimachinery/pkg/util/wait#Poll

Reply:
also is it possible to use `2*time.Second` to be coherent with `20*time.Second`?

Reply:
good point, that makes it more readable

Reply:
re: deprecation. We discussed about this with Brad in another PR https://github.com/k3s-io/k3s/pull/9076#discussion_r1434397387, I'll add a comment to the code

Reply:
The replacement `[PollUntilContextTimeout](https://github.com/kubernetes/apimachinery/blob/v0.29.1/pkg/util/wait/poll.go#L45)` was only added in 1.27, so we should continue to use the deprecated functions until 1.26 is end of life. Once we are no longer maintaining 1.26, we can switch to the new 1.27+ functions and backport that change to all the active branches.

Reply:
Created an issue to track: https://github.com/k3s-io/k3s/issues/9321

Reply:
Thanks for the explanation